addons/knowledge/models/knowledge_article.py

```diff
@@ -0,0 +1,424 @@
+                    vals['article_member_ids'] = [(0, 0, {
+                        'partner_id': member.partner_id.id,
+                        'permission': member.permission
+                    }) for member in parent.article_member_ids]
+
+        articles = super(Article, self).create(vals_list)
+        for article, vals in zip(articles, vals_list):
+            if any(field in ['parent_id', 'sequence'] for field in vals) and not self.env.context.get('resequencing_articles'):
+                article.with_context(resequencing_articles=True)._resequence()
+        return articles
+
+    def write(self, vals):
+        """ Add editor as author. Edition means writing on the body. """
+        if 'body' in vals:
+            vals.update({
+                "author_ids": [(4, self._uid)],  # add editor as author.
+                "last_edition_id": self._uid,
+                "last_edition_date": fields.Datetime.now(),
+            })
+
+        membership_changed = False
+        wrote_articles = self.env['knowledge.article']
+        if vals.get('parent_id'):
+            # copy members from parent
+            parent = self.browse(vals['parent_id'])
+            if not parent.user_can_write:
+                raise AccessError(_("You cannot move articles under an article you can't write on"))
+            vals['internal_permission'] = parent.internal_permission
+            for article in self:  # need to do it one by one as each article has their own set of members
+                if parent.article_member_ids:
+                    # In case of conflict, give priority to the article's members, not parent's ones
+                    article_vals = vals.copy()
+                    article_vals['article_member_ids'] = [(0, 0, {
+                        'partner_id': member.partner_id.id,
+                        'permission': member.permission
+                    }) for member in parent.article_member_ids if member.partner_id not in article.partner_ids]
+                    super(Article, article).write(article_vals)
+                    membership_changed = True
+                    wrote_articles |= article
+
+        result = super(Article, self - wrote_articles).write(vals)
+
+        # Propagate access rules to children
+        # TODO: /!\ What happen if you can write on parent article but not on one of the children ?
+        if self.child_ids:
+            children_values = {}
+            if 'internal_permission' in vals:
+                # Propagate permission to children
+                children_values = {
+                    'internal_permission': vals['internal_permission']
+                }
+
+            wrote_children = self.env['knowledge.article']
+            if 'article_member_ids' in vals or membership_changed:
+                # propagate members to children: need to do it one by one as each article has their own set of members
+                for article in self:
```

Comment:
Shouldn't we loop through all articles first? Or ensure there is only one? Won't this apply the changes to all articles' children, the ones that are not supposed to be changed?